### PR TITLE
Move secrets to keyvault instead of secrets.tfvars

### PR DESCRIPTION
**Pull request by kahlstrm**
This moves secrets from a private .tfvars file to an Azure Key Vault.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -260,3 +265,11 @@ module "invoicing" {
...
   dkim_key      = "k=rsa; p=MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAsAjLp5HRzTjMcjGGjZ75U42hCUaopuficYZiyLL3Ail/BmTqh2K8LRxN2UrWXOzVGLEh2F9PR6MC7nqR1Vj+3yR4+5nznwfmZh0cnX4Q2asm7A76st4uVwkVk0y21Mj1wufBIz885XCk+rzeorMOCU+lDZUIehYk1sVSDcubDuBAwJ9TBLXj2EMmcrD1KmJWMca0d5I6RfB+ZD7hG97rWhpgPuKYP7gaT6/t+ekXIJn9ZJmNRoIm/5X04AdM20ywwUrVe6NzWkB8eFuVy01DZki2bI9JnPwjnjw+KgZWrZBhtaYE8umVExmwGmI9PTzrHrknaBKQ0UBrDqSlyXuWgwIDAQAB"
 }
+
+module "keyvault" {
+  source   = "./modules/keyvault"
+  env_name = "prod"
+
+  resource_group_name     = module.common.resource_group_name
+  resource_group_location = local.resource_group_location
+}
```
Comment (PurkkaKoodari) [inline main.tf:275]: Might be clearer to put this first since everything depends on it.

File: variables.tf
Comment (PurkkaKoodari) [inline variables.tf:1]: image

File: modules/keyvault/main.tf [hunk 2]
```diff
@@ -0,0 +1,90 @@
+data "azurerm_client_config" "current" {}
+
+resource "azurerm_key_vault" "keyvault" {
+  name                        = "tikweb-keyvault-${var.env_name}"
+  location                    = var.resource_group_location
+  resource_group_name         = var.resource_group_name
+  enabled_for_disk_encryption = true
+  tenant_id                   = data.azurerm_client_config.current.tenant_id
+  soft_delete_retention_days  = 7
+  purge_protection_enabled    = false
```
Comment (PurkkaKoodari) [inline modules/keyvault/main.tf:10]: Do we want more protection against mistakes? `prevent_destroy` might also make sense.

**Review by PurkkaKoodari — APPROVED** [review]
LGTM. We might also want to document how this would be initially set up by a user.

I wonder if this will result in a lot of "will be read during apply" spam in every upcoming `plan`...

**Comment by kahlstrm** [comment]
> LGTM. We might also want to document how this would be initially set up by a user.
> 
> I wonder if this will result in a lot of "will be read during apply" spam in every upcoming `plan`...

I think those happen only when `# (depends on a resource or a module with changes pending)` i.e. if a resource that is being changed is dependent on a keyvault resource, only _then_ those are read. But not 100% certain

**Pull request merged**